.github/workflows/analyze.yml

```diff
@@ -0,0 +1,41 @@
+name: Analyse Bundle
+on:
+  pull_request:
+    branches:
+    - master
+  push:
+    branches:
+    - master
+
+jobs:
+  build_and_test:
+    name: Analyze Bundle
+    runs-on: Runner_16cores_Deriv-app
+    environment: Preview
+    steps:
+    - name: Checkout
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+    - name: Setup Node
+      uses: "./.github/actions/setup_node"
+    - name: Install dependencies
+      uses: "./.github/actions/npm_install_from_cache"
+    - name: Analyze
+      uses: "./.github/actions/analyze"
```

Comment:
just a suggestion, since we have build step as a separate actions isn't it better to reuse that action instead of adding a new action which exactly does the same?

Reply:
<img width="838" alt="timings" src="https://github.com/binary-com/deriv-app/assets/129744061/23bb7ea3-8eb2-49ca-8c8f-4127c1c0e6f4">

Hey Ako. Yeah it's a good point and something I tried too. The thing is, the build already takes 20 minutes, and if I add this analyse step, it adds an extra 8 minutes to the build.

If we run them in parallel, then while that work still needs to be done, it at least doesn't increase the build time.

Reply:
You don't need to run it in parallel or make any additions to the build. Instead, you can invoke the build workflow just as we do in the production release workflow. Then, utilize the bundles generated from that step. This way, if we modify the build workflow, the changes will automatically apply to all workflows that depend on it.
just like this: https://github.com/binary-com/deriv-app/blob/54a3356b80218274a488da6033e0b06a9bd32963/.github/workflows/release_production.yml#L26-L44

Reply:
Ahh, so if you look in production right now the build step is generating the stats.json file. The problem is, if the build fails then it swallows the errors. You can see this now on PR's if a build fails it just silently quits.

I think it's because it pipes to the stats.json file `> stats.json` so the errors are probably swallowed into that file.

Reply:
got it, then looks good to me